js/base/Exchange.js

```diff
@@ -2360,6 +2360,14 @@ module.exports = class Exchange {
         return this.createOrder(symbol, 'market', side, amount, undefined, query);
     }
 
+    checkOrderTypeAndPrice (market, type, side, amount, price) {
+        if (price === undefined) {
```

Comment:
Can we change the order of this condition  (check type first) or even merge them into one? Doing that if the type = market you don't need to proceed with the second check. 

Example: (since && is a lazy operator if the left side fails won't check the second condition)
```Javascript
if (time === 'limit' && price === undefined) {
    throw ....
}
```

Reply:
i thought that comparing to string was slower, and that's why i've thought it in opposite. it's also true in reverse - if price is set, it wont check the second condition.
but if you still think it's better, feel free to change, I'm aligned with you

Reply:
@ttodua Yeah, probably it is not that relevant so don't worry 🙂